### ISSUE TITLE
UI Theme Improvement

### ISSUE THREAD

**Issue by HugoPlacer** [drwhut/tabletop-club#119]
**Is your feature request related to a problem? Please describe.**
No.

**Describe the solution you'd like**
I think the UI theme should be other than Godot's default theme. Something simple, clean, and modern that match with the project.

**Describe alternatives you've considered**
Nothing.

**Additional context**

**Comment by drwhut** [comment]
Yeah, I agree! I hope you don't mind, but I'll probably take the opportunity to turn this thread into a "theme proposals" thread, so anyone can submit themes for the game.

If you have created a new theme for the game that you think fits, then feel free to post some screenshots using the theme of the following:
- The main menu.
- The "Video" tab of the options menu.
- The in-game UI without any windows open.
- The "Objects" window.
- The "Games" window.
- The "Notebook" window.

The reason being that the above items should showcase most (if not all) of the different UI elements in the game.

**Comment by HugoPlacer** [comment]
I made a Theme for the game, is in very early stage, and it's missing some themes, but I want some feedback to continue with the idea or start it over with other idea :).

Main Menu:
![Screenshot_20221203_131327](https://user-images.githubusercontent.com/69766683/205440711-b3100852-b82e-4522-a127-dcaefe21ad92.png)

"Video" tab:
![Screenshot_20221203_131408](https://user-images.githubusercontent.com/69766683/205440734-c13dd1f2-30d5-4fbc-b3cb-c31320e77961.png)

in-game UI:
![Screenshot_20221203_131442](https://user-images.githubusercontent.com/69766683/205440744-5e4f1ea1-8584-4bd8-871e-6a6139564b68.png)

"Objects" window:
![Screenshot_20221203_131500](https://user-images.githubusercontent.com/69766683/205440774-5ec86c79-2557-487b-894a-bdd2e4db5b14.png)

"Games" window:
![Screenshot_20221203_131516](https://user-images.githubusercontent.com/69766683/205440789-0095fffa-3fea-44c5-97c3-fbebf2872563.png)

"Notebook" window:
![Screenshot_20221203_131532](https://user-images.githubusercontent.com/69766683/205440798-0f18e980-20b7-40e5-bd20-89c28e625113.png)

**Comment by drwhut** [comment]
Not bad at all! I've compiled a few points for the bits that have changed:

- For buttons with icons in them, I think the margin between the icon and the border of the button is a bit too small.
- The tabs at the top of the options menu should be separated a bit I think.
- The backgrounds of buttons and panels seem a bit... dark? Maybe there could be a little bit of colour added, just to make the UI feel a bit more "cosy".
- For the tools on the left when in-game, I know the cursor tool is selected, but it kinda looks like all of the others are selected, and the cursor tool is de-selected due to the light border.

**Comment by HugoPlacer** [comment]
Thanks!!! You're right, :) I will try to change it :D. In the case of tabs, I have not been able to separate it :(.

**Comment by HugoPlacer** [comment]
I can't make the button icons smaller. They adjust to the left (I changing it with HSeparation option in theme).
![Peek 03-12-2022 16-45](https://user-images.githubusercontent.com/69766683/205449461-27722cb2-093f-4fd1-8f1a-24abe3bae253.gif)

**Comment by Jowan-Spooner** [comment]
Here are some of my suggestions regarding the in-game UX. They are based on these goals:
- Ease of Use
- Minimizing Info/Options
- Consistent Design Language

Here are the changes I would make:
- Remove `Flip Table`, `Clear Table`, move them in a drop down menu for various rarely used actions.
- Make `Objects`, `Games`, `Room` and `Notebook` popups NOT cover the whole screen. These would be so much nicer to use and more useful if you could still see the game.
- Allow hiding the whole Control-Tip stuff
- Use Icons only for all on-screen buttons (but add tooltips to all buttons)

Here is a mockup of how this could look:
![grafik](https://github.com/user-attachments/assets/0afdc37d-4995-4037-995a-0b569bccff9e)
![grafik](https://github.com/user-attachments/assets/c3dd2e0a-f408-4c54-bbe5-d07c59912275)
![grafik](https://github.com/user-attachments/assets/95c3ee9a-2be6-4ac6-9a23-d29e7559bccd)

**Comment by drwhut** [comment]
Hey! Sorry for the late reply, I've been extremely busy as of late.

> * Remove `Flip Table`, `Clear Table`, move them in a drop down menu for various rarely used actions.

I agree with this - In the `0.2-rewrite` branch I'm planning to move these buttons into both the right-click menu when you click the table, as well as in the "Room" window.

> * Make `Objects`, `Games`, `Room` and `Notebook` popups NOT cover the whole screen. These would be so much nicer to use and more useful if you could still see the game.

I agree that they could be smaller, but based on your mock-up, I still think it makes sense to have them be windows that can be manually moved, and that appear in the center of the screen by default. Not only because it allows the user to be able to move the window to where it is most convenient, but also for players with ultra-widescreen monitors, having them appear in the center of the screen where their focus is mainly going to be, instead of all the way to one side, makes sense in my opinion.

> * Allow hiding the whole Control-Tip stuff

This is already possible, although not as a button in the UI, but in the options menu: `Options` > `General` > `Hide Control Hints`.

> * Use Icons only for all on-screen buttons (but add tooltips to all buttons)

Hmm... I'm trying to think about this from a brand new player's perspective - I think for some of the buttons it may still be worth having text on them, depending on how important they are (for example, the "Objects" or "Games" buttons). That way, if only a select few buttons have text, new player's eyes will be drawn towards the buttons that they most likely want to use. Will have a think about this for v0.2.0, but obviously depending on feedback this can easily be adjusted.

In terms of the mock-up screenshots you gave, here's my general feedback:

* The colour scheme is really nice! Although, I'm still undecided if I want the UI to have a minimalist style to it, or whether it should have some other characteristic. I'm more focused on functionality than looks currently, but in the future I think it would be nice for the game to have a sort of cosy, jazz-club vibe to it, but I'm not sure how the in-game UI could reflect this to be honest.
* I'm not 100% sure on the tools being at the top, and the windows being on the right-hand side. This may just be because I want to try and somewhat replicate the layout that Tabletop Simulator has, so that the transition from that game to this one is as easy as possible (after all, I'm pretty confident that the majority of players have played something like Tabletop Simulator before this). But also, in most desktop applications, for example Photoshop, dialogs and windows tend to be along the top (or top-left, to be more accurate), and tools tend to be down the left-hand side.
* Having a button to hide the control hints on the bottom-right makes sense, plus it makes for really nice symmetry with the chat button on the other side, but given that it is already an option in the options menu, I need to think about how this would be implemented.

**Comment by drwhut** [comment]
I must stress, these screenshots are very, *very* work-in-progress, but this is currently what the new UI looks like on the `0.2-rewrite` branch, feedback is very much welcome!

**Menu Bar:**
![Screenshot from 2024-09-30 13-18-16](https://github.com/user-attachments/assets/6f86fae9-ea90-47df-bf87-53489db06c98)

**Menu Bar (Controller):**
![Screenshot from 2024-09-30 13-18-31](https://github.com/user-attachments/assets/70ce5731-1733-4965-9283-e95073656141)

**Objects Menu (Icons TBA):**
![Screenshot from 2024-09-30 13-18-58](https://github.com/user-attachments/assets/87444cdc-c952-4d51-b806-3ae732e912cc)

**Changing Objects Menu Appearance:**
![Screenshot from 2024-09-30 13-19-19](https://github.com/user-attachments/assets/d68962e6-0d62-4b0c-858a-bac2f17f7623)
